packages/vats/src/core/chain-behaviors.js

```diff
@@ -315,6 +317,40 @@ export const makeChainStorage = async ({
   chainStorageP.resolve(rootNodeP);
 };
 
+/** @param {BootstrapPowers} powers */
+export const publishAgoricNames = async ({
+  consume: { agoricNamesAdmin, board, chainStorage: rootP },
+}) => {
+  const root = await rootP;
+  if (!root) {
+    console.warn('cannot publish agoricNames without chainStorage');
+    return;
+  }
+  const nameStorage = E(root).getChildNode('agoricNames');
+  const marshaller = E(board).getPublishingMarshaller();
+
+  // brand, issuer, ...
+  await Promise.all(
+    keys(agoricNamesReserved).map(kind =>
```

Comment:
this indentation is pretty extreme. consider defining some of these callback functions passed into `then`

Reply:
yeah; this is a reflection of muddled thinking about whether the inner promises were going to resolve in time.